tests/testsuite/check.rs

```diff
@@ -1265,19 +1359,28 @@ fn check_fixable_error_no_fix() {
         )
         .build();
 
-    let rustc_message = raw_rustc_output(&foo, "src/lib.rs", &[]);
-    let expected_output = format!(
-        "\
-[CHECKING] foo v0.0.1 ([..])
-{}\
-[WARNING] `foo` (lib) generated 1 warning
-[ERROR] could not compile `foo` (lib) due to 1 previous error; 1 warning emitted
-",
-        rustc_message
-    );
     foo.cargo("check")
         .with_status(101)
-        .with_stderr(expected_output)
+        .with_stderr_data(str![[r#"
+[CHECKING] foo v0.0.1 ([ROOT]/foo)
+[ERROR] traits in `#[derive(...)]` don't accept arguments
+ --> src/lib.rs:2:15
+  |
+2 | #[derive(Debug(x))]
+  |               ^^^ [HELP] remove the arguments
+
+[WARNING] unused import: `std::io`
+ --> src/lib.rs:1:5
+  |
+1 | use std::io;
+  |     ^^^^^^^
+  |
+  = [NOTE] `#[warn(unused_imports)]` on by default
+
+[WARNING] `foo` (lib) generated 1 warning
+[ERROR] could not compile `foo` (lib) due to 1 previous error; 1 warning emitted
+
+"#]])
```

Comment:
Note: This change removes tests with messages originating from `raw_rustc_output` and directly compares them with snapshots. Please let me know if this approach is inappropriate.

Reply:
I feel like eventually raw_rustc_output can be removed. We should probably omit rustc diagnostics and only asset error code part.